packages/flutter_tools/lib/src/build_runner/web_fs.dart

```diff
@@ -236,29 +220,77 @@ class WebFs {
         }
       };
     });
+
     Handler handler;
     Dwds dwds;
-    if (!skipDwds) {
-      dwds = await dwdsFactory(
-        hostname: hostname ?? _kHostName,
-        applicationPort: hostPort,
-        applicationTarget: kBuildTargetName,
-        assetServerPort: daemonAssetPort,
-        buildResults: filteredBuildResults,
-        chromeConnection: () async {
-          return (await ChromeLauncher.connectedInstance).chromeConnection;
-        },
-        reloadConfiguration: ReloadConfiguration.none,
-        serveDevTools: true,
-        verbose: false,
-        enableDebugExtension: true,
-        logWriter: (dynamic level, String message) => printTrace(message),
-      );
-      handler = pipeline.addHandler(dwds.handler);
+    BuildDaemonClient client;
+    StreamSubscription<void> firstBuild;
+    if (buildInfo.isDebug) {
+      final bool hasWebPlugins = findPlugins(flutterProject)
+          .any((Plugin p) => p.platforms.containsKey(WebPlugin.kConfigKey));
+      // Start the build daemon and run an initial build.
+      client = await buildDaemonCreator
+        .startBuildDaemon(fs.currentDirectory.path,
+            release: buildInfo.isRelease,
+            profile: buildInfo.isProfile,
+            hasPlugins: hasWebPlugins,
+            initializePlatform: initializePlatform,
+        );
+      client.startBuild();
+      // Only provide relevant build results
+      final Stream<BuildResult> filteredBuildResults = client.buildResults
+        .asyncMap<BuildResult>((BuildResults results) {
+          return results.results
+            .firstWhere((BuildResult result) => result.target == kBuildTargetName);
+        });
+      firstBuild = client.buildResults.listen((BuildResults buildResults) {
```

Comment:
Is there a difference between "first" and "initial" builds?

Reply:
Fixed the names so that it is clear they are related